File: code-examples/pymongo/indexes/geo.py
# Create a geospatial index

from pymongo import MongoClient

# Replace the placeholders with your MongoDB deployment's credentials
uri = 'mongodb+srv://<user>:<password>@<cluster-url>?writeConcern=majority'

client = MongoClient(uri)

try:
    # begin-idx
    database = client["sample_mflix"]
    movies = database["movies"]

    # Create a 2dsphere index on the "location.geo" field in the
    # "movies" collection
    result = movies.create_index([("location.geo", "2dsphere")])

    # Print the result of the index creation
    print(f'Index created: {result}')
    # end-idx

finally:
    # Close the connection after the operation completes
    client.close()

File: code-examples/java-rs/indexes/geo.java
import com.mongodb.reactivestreams.client.MongoClients;
import com.mongodb.reactivestreams.client.MongoClient;
import com.mongodb.reactivestreams.client.MongoCollection;
import com.mongodb.reactivestreams.client.MongoDatabase;
import org.bson.Document;
import org.bson.conversions.Bson;
import org.reactivestreams.Publisher;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.ExecutionException;

public class CreateGeospatialIndex {

    public static void main(String[] args) throws ExecutionException, InterruptedException {
        // Connect to MongoDB
        MongoClient mongoClient = MongoClients.create("mongodb+srv://<user>:<password>@<cluster-url>?writeConcern=majority");
        
        {
            // Get database and collection
            MongoDatabase database = mongoClient.getDatabase("sample_mflix");
            MongoCollection<Document> collection = database.getCollection("movies");
            
            // Create index
            Bson index = new Document("location.geo", "2dsphere");
            Publisher<String> publisher = collection.createIndex(index);
            
            // Print index creation result
            CompletableFuture<String> future = new CompletableFuture<>();
            publisher.subscribe(new PrintDocumentSubscriber<>(future));
            System.out.println("The index was created with the following name: " + future.get());
        }
        
        // Close the connection
        mongoClient.close();
    }
}
